src/main/java/org/jboss/ejb/protocol/remote/EJBClientChannel.java

```diff
@@ -1320,13 +1320,10 @@ class MethodCallResultProducer implements EJBReceiverInvocationContext.ResultPro
              */
             private void cleanContextDataBeforeUnmarshalling(EJBClientInvocationContext clientInvocationContext) {
                 Map<String, Object> contextData = clientInvocationContext.getContextData();
-                @SuppressWarnings("unchecked")
-                Set<String> returnedContextDataKeys = (Set<String>) contextData.get(EJBClientInvocationContext.RETURNED_CONTEXT_DATA_KEY);
-                if(returnedContextDataKeys != null) {
-                    contextData.keySet().removeAll(returnedContextDataKeys);
-                }
+                contextData.keySet().removeIf(k -> (k != EJBClientInvocationContext.RETURNED_CONTEXT_DATA_KEY));
```

Comment:
This method needs clear doc stating what it does. In your latest change above, it now removes all entries from context data, except the one entry that specifies the `jboss.returned.keys`, and its value is a set of string keys. Is this what's intended here?  This is not what the javadoc says.

Should we use equals instead of != for comparison?